src/Features/Core/Portable/Completion/CommitChange.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Completion
+{
+    internal sealed class CommitChange
+    {
+        /// <summary>
+        /// The changes to be applied to the document.
+        /// </summary>
+        public ImmutableArray<TextChange> TextChanges { get; }
+
+        /// <summary>
+        /// The new caret position after commit.
+        /// If null then the caret position is not specified.
+        /// </summary>
+        public int? NewPosition { get; }
+
+        /// <summary>
+        /// True if the changes include the commit key.
+        /// </summary>
+        public bool IncludesCommitKey { get; }
+
+        private CommitChange(ImmutableArray<TextChange> textChanges, int? newPosition, bool includesCommit)
+        {
+            this.TextChanges = textChanges.IsDefault ? ImmutableArray<TextChange>.Empty : textChanges;
+            this.NewPosition = newPosition;
+            this.IncludesCommitKey = includesCommit;
+        }
+
+        public static CommitChange Create(ImmutableArray<TextChange> textChanges, int? newPosition = null, bool includesCommitKey = false)
+        {
+            return new CommitChange(textChanges, newPosition, includesCommitKey);
+        }
+
+        public CommitChange WithTextChanges(ImmutableArray<TextChange> changes)
+        {
+            return new CommitChange(changes, this.NewPosition, this.IncludesCommitKey);
+        }
+
+        public CommitChange WithNewPosition(int? position)
```

Comment:
this name should be "newPosition".  Especially in the new immutable world, we want parameternames/propertynames to be consistent.